src/engine/strat_engine/dmdevice.rs

```diff
@@ -0,0 +1,142 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Functions for dealing with device mapper devices.
+
+use std::fmt;
+use std::fmt::Display;
+
+use super::super::super::engine::{EngineResult, EngineError, ErrorEnum, PoolUuid};
+
+const FORMAT_VERSION: u16 = 1;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Layer {
+    Flex,
+    Thin,
+    ThinPool,
+}
+
+impl Display for Layer {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Layer::Flex => write!(f, "flex"),
+            Layer::Thin => write!(f, "thin"),
+            Layer::ThinPool => write!(f, "thinpool"),
+        }
+    }
+}
+
+pub enum FlexRole {
+    MetadataVolume,
+    ThinData,
+    ThinMeta,
+}
+
+impl Display for FlexRole {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            FlexRole::MetadataVolume => write!(f, "mdv"),
+            FlexRole::ThinData => write!(f, "thindata"),
+            FlexRole::ThinMeta => write!(f, "thinmeta"),
+        }
+    }
+}
+
+pub enum ThinRole {
+    Filesystem,
```

Comment:
I think that the problem is that we _must_ disambiguate between different thin devices that Stratis might have set up, and the filesystem UUID is the only way to do it. This our Stratis filesystem UUID, not any UUID that the filesystem itself has.